### DISCUSSION TITLE
Keyword to automatically call parent class method

### DISCUSSION THREAD

**Discussion by Kelin2025** [godotengine/godot-proposals#5681] [Scripting]
For now only some of the internal methods are called in all parent classes
<img width="730" alt="Screenshot 2022-10-31 at 02 10 59" src="https://user-images.githubusercontent.com/4208480/198906422-72e87cf3-4167-4c5e-89ba-f3435a3370e3.png">
If you want to do the same with your own, you have to write boilerplate code like this:
```gdscript
func _applied(dealer, receiver, status):
	._applied(dealer, receiver, status)
	receiver.add_effect(stun)

func _removed(dealer, receiver, status):
	._removed(dealer, receiver, status)
	receiver.remove_effect(stun)
```
It would be cool to have something like `extra` keyword which does the same with less code:
```gdscript
extra func _applied(dealer, receiver, status):
	receiver.add_effect(stun)

extra func _removed(dealer, receiver, status):
	receiver.remove_effect(stun)
```
Thoughts?

---

By the way, the fact that __some of the methods__ are being called automatically is a source of big confusion. Spent 2 hours to figure out why my method doesn't get called in the parent class, because I was doing the same only with `_physics_process` before and thought it works the same way with all methods

**Comment by Calinou** [comment]
`super()` does this in 4.0.beta:

```gdscript
func _applied(dealer, receiver, status):
	super(dealer, receiver, status)
	receiver.add_effect(stun)

	# To call another function:
	# super.some_other_function(some_param)
```

> By the way, the fact that some of the methods are being called automatically is a source of big confusion. Spent 2 hours to figure out why my method doesn't get called in the parent class, because I was doing the same only with _physics_process before and thought it works the same way with all methods

This was changed in 4.0.beta as well. Multi-level calls must now be done explicitly with `super()` for all functions.

**Reply by Kelin2025** [reply]
Yeah but you have to pass arguments manually. In my example you don't

**Reply by Mickeon** [reply]
An annotation for this purpose doesn't sound too bad for this, actually.

It seems like the most common use case for super() is a call to the inherited class' method, right underneath the definition of the overriden method. So, a theoretical `@super` (where there's no need of annoying to both read and write boilerplate code) could be beneficial, and perhaps satisfying to the few users that find the "do not call some overridable methods" change an annoying one.
